go/vt/mysqlctl/grpcmysqlctlclient/client.go

```diff
@@ -108,6 +108,7 @@ func (c *client) withRetry(ctx context.Context, f func() error) error {
 		}
 		if err := f(); err != nil {
 			if grpc.Code(err) == codes.Unavailable {
+				lastError = err
```

Comment:
don't you want to assign lastError above `if err == Unavailable` condition? also maybe select between ctx.Done and time.Sleep below?